### DISCUSSION TITLE
Wagtail for static sites?

### DISCUSSION THREAD

**Discussion by Wysardry** [wagtail/wagtail#10635] [General]
I've looked around the Wagtail site and browsed the docs and have a few questions about using a local installation of Wagtail to create static HTML pages to upload to a host:-

Does Wagtail Bakery build all the pages in a site every time you use it, or does it only build pages that are new or changed (since the last build)?

Is there currently a way to connect Wagtail to a Python static site generator such as Nikola or Pelican?

Are the extensions from Python Markdown Extra supported?

Are there any issues with using Wagtail on Windows 7 or 10?

Finally, other than Bootstrap, what are the advantages of using the CodeRed extensions?

**Comment by RealOrangeOne** [comment]
This is far more than a single question, and would probably be better as a discussion in somewhere like Slack, but here goes:

[Wagtail Bakery](https://github.com/wagtail-nest/wagtail-bakery) Builds an entire site from scratch. It's meant more as a way of serializing an entire site than it is as part of a headless-style build. I'm sure it wouldn't be too difficult to submit a PR to add partial-update support.

Connecting to a static site is more a question for the static sites than Wagtail. Wagtail has a [robust REST API](https://docs.wagtail.org/en/stable/advanced_topics/api/index.html) and [supports GraphQL](https://github.com/torchbox/wagtail-grapple/). If your static site generator supports getting its content from an API, you're set. Wagtail is definitely ["headless"](https://areweheadlessyet.wagtail.org/)

Wagtail's markdown support is external to Wagtail. If the package you use supports adding markdown extensions, then it should all work as a custom streamfield block.

Developing Wagtail on Windows 7 and 10 should be fine, although we don't currently actively test against it, as most production environments run Linux. There are however definitely people developing on Windows 7 and 10 and we've heard no complaints.

I'm not very familiar with the CodeRed extensions, but I believe they are just that - extensions. Under the hood, it's all still core Wagtail.

With all this said, it sounds like you're trying to use Wagtail as part of a headless built static site. Whilst you can absolutely do this, there are significant extra complexities behind doing this (as you've noted by opening this discussion). I (personal opinion bsaed on experience) **highly** recommend not using headless, and instead developing a _conventional_ Wagtail application, and using something like a CDN to cache pages to reduce load times and server load. You can even [connect Wagtail to your CDN](https://docs.wagtail.org/en/stable/reference/contrib/frontendcache.html) so users see page changes as soon as they're available.

**Reply by zerolab** [reply]
Great summary, @RealOrangeOne 👏🏼 

fwiw, https://github.com/torchbox/wagtail-markdown/ supports various Markdown extensions

CodeRed extensions (or CRX) was formerly known as CodeRed CMS - https://github.com/coderedcorp/coderedcms and is "Wagtail + CodeRed Extensions enabling rapid development of marketing-focused websites.", that is, an opinionated Wagtail distribution

**Comment by Wysardry** [comment]
So Wagtail Bakery is where the partial update feature would need to be added (rather than Django Bakery)?

I'm unsure if any of the Python static site generators I've been looking at support REST or GRAPHQL API connections. Nikola does have plugins available to connect to Dato CMS, Netlify and Contentful, but the docs don't say much about how they were implemented.

I would only need to use a static site generator if it there was no way of getting Wagtail Bakery to build only new and changed pages.

The main priority for me is to build static HTML pages, rather than having an easy to use front end. Fairly obviously, I would _like_ the editing process to be as simple and easy as posible, but not by making the site itself more complicated than it needs to be.

**Comment by Wysardry** [comment]
In case anyone else might find it useful, I have found a tool that will only transfer files that have changed. It's called [FreeFileSync](https://freefilesync.org/).

I would still prefer that only new and changed files were processed, as it would be less wear and tear on my hard drive.
